### ISSUE TITLE
ZK-SJ30 30A dcdc model with better efficiency ?

### ISSUE THREAD

**Issue by RoboDurden** [RoboDurden/Esp32_20A_80V_dcdc#4]
There is a 30A version with the same LCD module, so it might be compatible with this project.

22,59€  39%OFF | ZK-SJ30 700W High Power Auto Boost Power Module Synchronous Rectifier 30A Solar Charging MPPT
https://a.aliexpress.com/_EHqaZRz

I am not satisfied with the efficiency of the 20A dcdc. At best I verified an efficiency of only 92%. A really boring video with lots of input output measurements that I do not think worth uploading to YouTube..

@I-hate-2FA , what would this 30A dcdc module would cost on these platforms ?

https://my.world.taobao.com/item/789873368119.htm

Usd $10 ? But what about shipping ??

https://www.ruten.com.tw/item/show?22420585284046

??

I will order two modules on AliExpress and test the efficiency. Hopefully they have better mosfets..

The one customer feedback sounds promising:

![Screenshot_20240618-173959](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/12238841/e191b666-7a41-42b9-b98d-9bfa9f4f44a2)

(I will finalize v3 in the next two weeks..)

**Comment by I-hate-2FA** [comment]
is this one also buck-boost topology or other
because the inductor is tiny for its power, i dont believe the specs at all
and what main chip is it using

the module cost $25usd with display or $20 without, idk abt the shipping, it is viewed with china version of the website

**Comment by RoboDurden** [comment]
@I-hate-2FA you will find more happiness if you think positive..

This 30A version has 8 mosfet instead of the normal 4 mosfet of the 20A version.
So half the resistance. That should indeed alow for more currents or better efficiency.

30A: https://wmsc.lcsc.com/wmsc/upload/file/pdf/v2/lcsc/2202132000_KIA-Semicon-Tech-KCB3008B_C2839423.pdf
RDS(on).typ= 4.5 mΩ , 85 V
20A: https://www.huanqiuic.com/uploads/pdf/20220108/pYYBAGG25B2APGWxAAWzSyBL_80872.pdf
RDS(on).typ= 4.3mΩ , 82 V

The 30A version has 12V regulator onboard available to the 2-pin fan socket, the 20A module has 5V.
Which i use to power the ESP32 and tft display. 
So my pickup module will need an an additional 5V regulator.

The two female headers seem to be compatible to my module. 
Yes there is 3.3V on the third pin of the right header. Which normally powers the lcd module.
So it might be possible to use this to power my esp32 and tft (which will be not as bright as 5V but still good to see.). 
But that would need a v4 of this pcb here.
And this 3.3V seem to come from a linear regulator. So from 11.40 (12V) down to 3.3V :-( Okay for the low power consumption of the original lcd module, but not really nice for wifi-esp32 + tft + i2c modules..

v3 will be delieverd in a few days. But i will assemble it in about 2-3 weeks i guess.

**Comment by RoboDurden** [comment]
![PXL_20240701_093422535_copy_1512x2016](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/12238841/62a0ac2e-fc54-452d-b4a3-a7fdb9aeb271)
![PXL_20240701_090527478_copy_2016x1512](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/12238841/bbbc748a-2e2f-4e65-8171-be954c34e217)
![PXL_20240701_090514277_copy_1639x1398](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/12238841/586c7d8d-b8a3-4e7b-a9d7-c5abdba32074)

I added the nice 14A ltc3780 stepup step down module for which I have verified 97% efficiency. 
14,29€ https://a.aliexpress.com/_EvPGhKj

So I do not think that the coil of the lt8705 modules are undersized.
They both have four wires in parallel.

**Comment by I-hate-2FA** [comment]
the module just look small on the listing, it is actually huge, way too large for my project
my portable psu recently broke down, is is very unstable, already burned 2 motherboard because the output went out of control, i really need to start building a new one

**Comment by RoboDurden** [comment]
I am satisfied with this stepdown module: https://youtu.be/fOnTM6UMRIU

**Comment by I-hate-2FA** [comment]
> I am satisfied with this stepdown module: https://youtu.be/fOnTM6UMRIU

the problem is i need buck boost one(work on 2~6s lipo voltage), and controlled by esp32

**Comment by I-hate-2FA** [comment]
the one you have can be controlled by esp with serial, but it will only output less then 12v which is not enough, ideally i want 50v, but 36v is also ok

**Comment by I-hate-2FA** [comment]
i also ideally want it to fit in standard powerbank size, it is clearly possible with current tech but not with readlly available module

**Comment by RoboDurden** [comment]
the module can go below 12V! Have quickly made a video for you: https://youtu.be/4G_sZVq6stk
But as it has a range of 1.2V - 80V, the potentiometer is very sensitive. But later in the video i demonstrate that it is also very sensitive for 30V, and for such voltages my v2 board with 10 bit software/filter dac works nicely.

**Comment by RoboDurden** [comment]
> i also ideally want it to fit in standard powerbank size, it is clearly possible with current tech but not with readlly available module

This sounds rather like a Kickstarter project then only for your personal use :-)

**Comment by I-hate-2FA** [comment]
> the module can go below 12V! Have quickly made a video for you: https://youtu.be/4G_sZVq6stk But as it has a range of 1.2V - 80V, the potentiometer is very sensitive. But later in the video i demonstrate that it is also very sensitive for 30V, and for such voltages my v2 board with 10 bit software/filter dac works nicely.

looks great, good for desktop power supply, but too large for portable one, and still need to figure out how to control feedback pin with micro controller, and not sure if the constant current is fast enough for a bench supply as its only meant for charge battery (if you set to 12v and connect 5mm led it wont explode then its good)

and i want to know, does the potentiometer have full 80v on it? because it will be operating out of specs, but its chinese product i wont be surprised if it is, if so it rules out any possibility of digital potentiometer

im curious how does comercial digitally controlled module have such good accuraicy, with max 50v and 0.001v resolution, what is the pwm frequency they used?

**Comment by RoboDurden** [comment]
![v3](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/12238841/505f891a-0f8d-487f-8682-995c61f2b3c5)

**Comment by I-hate-2FA** [comment]
> This sounds rather like a Kickstarter project then only for your personal use :-)

it is just for my personal use, but if it happens i will make a video (a showcase at lease) with all the design files if you want to build one, nowadays custom pcb and 3d printing just make it possible to be done at home, heck you can even order custom silicon now

**Comment by RoboDurden** [comment]
Well for personal use you mostly do not need a super small and compact design !
Only when you want to compete with other products and stand out as "the smallest portable power supply", would you need something not yet seen.
So if you do not see a success of a 100 pcs Kickstarter, it is probably not worth doing it.

Very strangely i tested the full range of the 30A for you and with this high sensitivity the output went to 96V with only 2 or three turns of the potentiometer.
Then something happened and the module changed to the nice slow sensitivity with 25 turns to achieve a range from 1.x Volt up to only 72 Volt.
And then it also had good control at the very low voltage regime.
With my 10 bit dac i think at least achive 0.1 Volt control easily.

If you want to go above 30 Volt, i think there currently is only see the LT8705 on the market.

**Comment by I-hate-2FA** [comment]
> Well for personal use you mostly do not need a super small and compact design !
> Only when you want to compete with other products and stand out as "the smallest portable power supply", would you need something not yet seen.

because i actually will need to bring it with me everywhere so i want to make it reasonably compact
there is already comercial mini power supply powered from usb pc with 100w 30v 5a 
https://www.youtube.com/watch?v=Pd6LG7iP2GQ
but i need more feature then that, so ill make my own



> Very strangely i tested the full range of the 30A for you and with this high sensitivity the output went to 96V with only 2 or three turns of the potentiometer.
> Then something happened and the module changed to the nice slow sensitivity with 25 turns to achieve a range from 1.x Volt up to only 72 Volt.

bad connection inside the potentiometer


> With my 10 bit dac i think at least achive 0.1 Volt control easily.

0.1v is good enough for most use case lets be real



> If you want to go above 30 Volt, i think there currently is only see the LT8705 on the market.

using generic pwm ic (or even let esp32 directly drive the mosfet)it can acheive arbatray voltage easily, but it is less stable then ready made ic which is tested for all condition
maybe 36v is enough, i really rarely use mine for such high voltage, but it is really good to have when you do, but itll be too large to fit in my project, even the 20a one

**Comment by RoboDurden** [comment]
This one is smaller (10x8x3,5 cm) and fits your cc/cv range: https://de.aliexpress.com/item/1005006320038436.html
You should be able to unsolder the digital pickup module and either generate the dac with esp32 or 12bit i2c module, or interface some dac chips on the main board.

**Comment by I-hate-2FA** [comment]
what topology is this, because it have 2 inductor, it is either sepic or inverted buck boost, and is it synchronus, because in a case without much space heating is a problem, i wont be using more then 200w but still, current module at 50w can already trigger otp

the ltc3780 may still be preferable
https://de.aliexpress.com/item/1005005301470472.html
or even the ltc1871, for the size of a bit larger then lm2596 you get over 100w with 95% effiency which is great for my project
https://de.aliexpress.com/item/1005004351702604.html

i have heard the ltc3780 module design is bad, but im not entirely sure how bad it really is, or they just use it out of specs as it does not have over power protection
but these module is not designed to have a digital control, so some more expermentation is required

**Comment by I-hate-2FA** [comment]
i really dont want inverted topology, because it is not isolated, and the usb port i will connect in parallel will not be on same level, it may easily go wrong

**Comment by RoboDurden** [comment]
Yes, i do not like this cheap ltc370 module with the three potentimeters. It gets hotter than the slightly older design with only output voltage control. On which i had build my 97% efficiency mppt: https://www.youtube.com/watch?v=1vDQG1ODD4M
I had a closer look on this other module and found out that the imput voltage control was simply done buy triggering the enable pin of the ltc3870. So NO analog feedback control :-(
I do not remember if this is also the case for the CC control loop.
But it got way hotter than the simple design.
The ltc1871 you link is only step up.

Yes, two coils looks strange. I guess they simply put a step up circuit in parallel to a step down circuit. Might be okay.

**Comment by I-hate-2FA** [comment]
this module is 60v,180w,small size
https://de.aliexpress.com/item/1005004964214245.html
but the downside is it is sepic and not synchronus, so the effiency is probably about 80%, it can probably melt pla case, at least sepic is not inverted(it have 2 coil on one ferrite)

i also have bought a similar one to this
https://de.aliexpress.com/item/1005005695046766.html
but it seems ic have no marking, and bad build quality(mosfet half floating in mid air, and not new mosfet), not sure about the topology, but i guess it is synchronus, not tested effiency tho

btw, i found a module with stm8 microcontroller, it is not any good though, only good for hackability
https://de.aliexpress.com/item/1005005710098471.html

**Comment by I-hate-2FA** [comment]
> Yes, i do not like this cheap ltc370 module with the three potentimeters. It gets hotter than the slightly older design with only output voltage control.

old one only rated for 24v not 36v

> I had a closer look on this other module and found out that the imput voltage control was simply done buy triggering the enable pin of the ltc3870. So NO analog feedback control :-(

input voltage control is not useful anyways, at least for me


> I do not remember if this is also the case for the CC control loop.

CC they used a opamp to somehow add current signal with voltage into the voltage feedback pin

**Comment by RoboDurden** [comment]
Yes i also have bought two of the 600W step up for my car booster but never tested them because i noticed that i need reverse currents when braking. This can only be done with step up/down.

The other two modules you have linked only seem to be step down.

**Comment by I-hate-2FA** [comment]
> The other two modules you have linked only seem to be step down.

180w one seller say is buck boost

**Comment by RoboDurden** [comment]
> old one only rated for 24v not 36v

Ah yes, you need to add a resistor to the CV potentiometer in order to increase possible output voltage to 30V and above.

The limit for both modules is the LTC3780 which has this 36V limit.
The old module simply was designed for 24V so the CV potentiometer needs that little tweak.

I have worked A LOT with this "old" module.

**Comment by RoboDurden** [comment]
> Input 10.5-70V  , Output 5-58V

That looks very much like step down only.

And the input capacitors are 80V whereas the output capacitors are 63V only.

But it is cheap - please test it :-)

**Comment by I-hate-2FA** [comment]
the reason allot of module have max 36v is if you need more a seperate small buck converter is needed to power the pwm chip, like the 180w one have



> That looks very much like step down only.

chinese description says it can increase,decrease voltage, and same input voltage as output, yes it can only output 58v but the input can be at 12v

**Comment by I-hate-2FA** [comment]
> But it is cheap - please test it :-)

ok, buy it locally is only 3x the price compared to buy from china, it is still acceptable

the seller linked a test video, i was gonna send it to you but it is for completely different module

**Comment by I-hate-2FA** [comment]
![image](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/166701841/cba22bc4-63d6-47ac-ace1-9c7b2cc2579b)
i have reverse engineered the schematic of the 180w one, it match a sepic converter, which can buck boost
![Screenshot_2024-07-02-21-28-43-627-edit_com taobao taobao](https://github.com/RoboDurden/Esp32_20A_80V_dcdc/assets/166701841/8066ecb2-d35b-4960-b438-8e1ac7b79ea8)

**Comment by RoboDurden** [comment]
I will not sponsor such a dcdc module, you have to buy it yourself. 
This is not really about making the world a better place.
But nice reverse engenieering !
Maybe i will order one as well to test efficiency. But it is rated for 30 Watt only without the heat sink. So It might perform poorly anyway.. https://de.aliexpress.com/item/1005002910548414.html

**Comment by I-hate-2FA** [comment]
> I will not sponsor such a dcdc module, you have to buy it yourself.

it is ok, i already said i will

> Maybe i will order one as well to test efficiency. But it is rated for 30 Watt only without the heat sink. So It might perform poorly anyway.

it is not synchronus, effiency always worse

**Comment by I-hate-2FA** [comment]
if you want high effiency, small size, high power, this is the best one you can buy
https://de.aliexpress.com/item/1005007259645809.html
based on lt8390 or isl81601, but it is too expensive

**Comment by RoboDurden** [comment]
Yes too expensive for me.
Bye for now, i want to focus on my new diy boiler heater i make out of 16x 60cm carbon/silikon heat wires: https://de.aliexpress.com/item/1005005192378143.html

**Comment by I-hate-2FA** [comment]
i am working on replace appliance motherboard with esp32 running esphome

**Comment by I-hate-2FA** [comment]
i have ordered both the 180w sepic converter and the 1000w full bridge buck boost converter thats way overpriced
lets see when they will arrive

**Comment by RoboDurden** [comment]
I don't think that these will be useful for my project here.

**Comment by I-hate-2FA** [comment]
but now i think about it, there is a much better solution
we could just use a buck boost converter chip that is intended to be controlled by a micro controller
like for example this one
https://cn.southchip.com/upload/file/SC8806_datasheet_brief_en.pdf
it is full bridge synchrnous, with external nmos so you can increase the current capability as much as you like, and directly use i2c to control the output voltage and current by hardware, with included protection feature
even better is this chip support invert input and output, so you can use it to charge a battery

**Comment by RoboDurden** [comment]
i do not feel experienced enough to design a dcdc module from scratch. And this chip can only handle up to 30V :-( 
Nor is it available on lcdc or mouser.

You remind me of myself when i was 20 years younger. A dreamer who always wants to create the perfect thing.
I strongly recomend that you watch Fight Club - at least ten times.

perfection is for failures. 

roland
"you either lose.. or fail - nothing else, nothing in between."

**Comment by I-hate-2FA** [comment]
> i do not feel experienced enough to design a dcdc module from scratch. 

just order one from jlcpcb
https://oshwhub.com/Monnina/SC8721QFERfang-an-yan-zheng-ban
and ofc you can download the pcb file and edit it if needed

> And this chip can only handle up to 30V :-(

these chip is designed for usb pd so unfortunately yes it will have the 36v hard limit, we are going to misuse it for bench supply and battery charger

> Nor is it available on lcdc or mouser.

yeah not available on lcsc yet, mouser is just trash it dont even have the ch32v003 yet, it will never have good things

**Comment by I-hate-2FA** [comment]
with usb pd 4.0 which need 48v, we are expecting 60v chips to come in about 2-4 years

**Comment by I-hate-2FA** [comment]
https://oshwhub.com/killy.thi/SC8815-200W-Bidirectional-BuckBoost
the one i sent the chip have internal switch, this one have external one(also i2c controlled)

**Comment by RoboDurden** [comment]
Sure, when such a module is available on AliExpress for $20- i might like to make such a pickup module for this like i am trying to do here for ZK-SJ20 module. But still there is the 30V limit, which makes it pretty much useless for solar charging or hoverboards

**Comment by I-hate-2FA** [comment]
![IMG_20240730_203917](https://github.com/user-attachments/assets/c4f2a8a9-736e-42ca-94c4-67e8947ea220)
![IMG_20240730_203857](https://github.com/user-attachments/assets/308f94db-0edf-4af6-811b-02abbaf9bafb)

**Comment by I-hate-2FA** [comment]
180w one can buck boost I tested
